actors/miner/tests/util.rs

```diff
@@ -204,8 +261,1024 @@ impl ActorHarness {
+        let dln = self.get_deadline(rt, deadline.index);
+        let post = self.get_submitted_proof(rt, &dln, proof_index);
+
+        for idx in post.partitions.iter() {
+            let partition = self.get_partition_snapshot(rt, &dln, idx);
+            all_ignored |= &partition.faults;
+            assert!(partition.recoveries.is_empty());
+        }
+
+        // find the first non-faulty, non-skipped sector in poSt to replace all faulty sectors.
+        let mut maybe_good_info: Option<SectorOnChainInfo> = None;
+        for ci in infos {
+            if !all_ignored.get(ci.sector_number) {
+                maybe_good_info = Some(ci.clone());
+                break;
+            }
+        }
+        let good_info = maybe_good_info.unwrap();
+
+        let entropy = RawBytes::serialize(self.receiver.clone()).unwrap();
+        rt.expect_get_randomness_from_beacon(
+            DomainSeparationTag::WindowedPoStChallengeSeed,
+            deadline.challenge,
+            entropy.to_vec(),
+            challenge_rand.clone(),
+        );
+
+        let mut proof_infos = Vec::new();
+        for ci in infos {
+            let mut si = ci.clone();
+            if all_ignored.get(ci.sector_number) {
+                si = good_info.clone();
+            }
+            proof_infos.push(SectorInfo {
+                proof: si.seal_proof,
+                sector_number: si.sector_number,
+                sealed_cid: si.sealed_cid,
+            });
+        }
+
+        let vi = WindowPoStVerifyInfo {
+            randomness: challenge_rand.clone(),
+            proofs: post.proofs,
+            challenged_sectors: proof_infos,
+            prover: RECEIVER_ID,
+        };
+        let verify_result = match expect_success {
+            Some(_) => ExitCode::ErrIllegalArgument,
+            None => ExitCode::Ok,
+        };
+        rt.expect_verify_post(vi, verify_result);
+
+        if expect_success.is_some() {
+            let dispute_result = expect_success.clone().unwrap();
+
+            if dispute_result.expected_power_delta.is_some() {
+                let expected_power_delta = dispute_result.expected_power_delta.unwrap();
+                let claim = UpdateClaimedPowerParams {
+                    raw_byte_delta: expected_power_delta.raw,
+                    quality_adjusted_delta: expected_power_delta.qa,
+                };
+                rt.expect_send(
+                    *STORAGE_POWER_ACTOR_ADDR,
+                    PowerMethod::UpdateClaimedPower as u64,
+                    RawBytes::serialize(claim).unwrap(),
+                    TokenAmount::from(0),
+                    RawBytes::default(),
+                    ExitCode::Ok,
+                );
+            }
+
+            if dispute_result.expected_reward.is_some() {
+                let expected_reward = dispute_result.expected_reward.unwrap();
+                rt.expect_send(
+                    self.worker.clone(),
+                    0, // builtin0.MethodSend
```

Comment:
same nit: you can replace magic number and comment with import of METHOD_SEND

Reply:
ah yes, of course; will do.

Reply:
done.